forc-pkg/src/pkg.rs

```diff
@@ -399,17 +400,33 @@ impl BuildPlan {
     // the manifest alongside some lock diff type that can be used to optionally write the updated
     // lock file and print the diff.
     pub fn from_lock_and_manifest(
-        manifest: &PackageManifestFile,
+        manifests: &ManifestFiles,
         locked: bool,
         offline: bool,
     ) -> Result<Self> {
```

Comment:
Ahh, I meant to add a comment here as a part of the last review but looks like it didn't make it:

It looks like the `WorkspaceManifestFile` is solely used to determine the lock file path. Instead, I think it might be better if we can pass the lock path as an argument to this constructor, e.g.

```rust
pub fn from_lock_and_manifest(
    lock_path: &Path,
    manifests: &MemberManifestFiles,
    locked: bool,
    offline: bool,
) -> Result<Self> {
```

This way the `BuildPlan` construction can remain agnostic of whether or not it's for a workspace.